msgraphcore/middleware/utils.py

```diff
@@ -0,0 +1,31 @@
+import datetime
+from email.utils import parsedate_to_datetime
+
```

Comment:
These are likely to only be used by retry_middleware so it may be better to add these to retry_middleware.py instead of a separate file.

Reply:
Updated the changes to reflect this